apps/slackbot/task_agent.py

```diff
@@ -150,6 +152,29 @@ def run(self, task: str) -> str:
             action = self.output_parser.parse(assistant_reply)
             print("action:", action)
             tools = {t.name: t for t in self.tools}
+            if action == previous_action:
+                if action.name == "Search" or action.name == "Context Search":
+                    print("Action name: ", action.name, "\nStart reformulating the query")
+                    instruction = (
+                        f"You want to search for useful information to answer the query: {task}."
+                        f"The original query is: {action.args['query']}"
+                        f"Reformulate the query so that it can be used to search for relevant information."
+                        f"Only return one query instead of multiple queries."
+                        f"Reformulated query:\n\n"
+                    )
+                    openai.api_key = environ.get("OPENAI_KEY")
+                    response = openai.Completion.create(
```

Comment:
We may want to use `self.llm` so that we don't need to provide extra configurations of the LLM here